### PR TITLE
ast: check annotations use current major version

### PR DESCRIPTION
**Pull request by kitbellew**
This way, we'll remember to remove older field annotations before a major version release (which allows binary non-compatibility).

### CODE REVIEW

File: scalameta/common/shared/src/main/scala/scala/meta/internal/trees/ast.scala [hunk 1]
```diff
@@ -459,15 +459,23 @@ class AstNamerMacros(val c: Context) extends Reflection with CommonNamerMacros {
...
+
+  private def parseVersionAnnot(version: Tree, annot: String, field: String): Version = {
+    val parsed =
+      try { parseVersion(getAnnotAttribute(version)) }
+      catch {
+        case NonFatal(_) => c.abort(version.pos, s"@$annot must contain $field=major.minor.patch")
+      }
+    majorVersion.foreach { major =>
+      if (parsed._1 < major)
+        c.abort(version.pos, s"@$annot: invalid major version (must be $major)")
```
Comment (tgodzik) [inline scalameta/common/shared/src/main/scala/scala/meta/internal/trees/ast.scala:476]: Shouldn't the message say something like `@$annot: version for $field no longer valid, please update` ?

Comment (kitbellew) [reply scalameta/common/shared/src/main/scala/scala/meta/internal/trees/ast.scala:476]: so, i don't want the version to be updated :) my goal is to make sure that all these binary-compatibility annotations are removed in a new major release (which will also result is a binary incompatibility, but that's what major releases are for). should i revise the message a bit, to say that?

Comment (kitbellew) [reply scalameta/common/shared/src/main/scala/scala/meta/internal/trees/ast.scala]: added the word `obsolete`, perhaps that makes it clearer :)

Comment (tgodzik) [reply scalameta/common/shared/src/main/scala/scala/meta/internal/trees/ast.scala]: Sounds better, thanks!

**Pull request merged**